lib/saml_config.rb

```diff
@@ -2,12 +2,10 @@
 
 module Saml
   class Config
-    def initialize
-      @@settings ||= build_settings.freeze
-    end
+    attr_reader :settings
 
-    def settings
-      @@settings
+    def initialize
+      @settings ||= build_settings.freeze
```

Comment:
I think there are some wires crossed? @jessieay that blog post is talk about instance vars on a class object, I think @pkarman is talking about instance variables on an instance of a class.

We instantiate `Saml::Config` objects like this in at least 4 spots, two are inside of controllers, so they are per-request.

```
identity-dashboard:master> git grep Saml::Config.new -- {app,config} | wc -l
       4
```

So I think what @pkarman is suggesting is

```ruby
module Saml
  class Config
    cattr_accessor :settings do
      build_settings # promote this to a class method
    end

    def initialize
      @settings = self.class.settings.dup.freeze
    end
  end
end
```